webknossos-datastore/app/com/scalableminds/webknossos/datastore/models/datasource/DataLayer.scala

```diff
@@ -66,9 +66,11 @@ trait DataLayerLike {
 
   def resolutions: List[Point3D]
 
-  def lookUpResolution(resolutionExponent: Int): Point3D = {
+  def lookUpResolution(resolutionExponent: Int, snapToClosest: Boolean = false): Point3D = {
     val resPower = Math.pow(2, resolutionExponent).toInt
-    resolutions.find(resolution => resolution.maxDim == resPower).getOrElse(Point3D(resPower, resPower, resPower))
+    val matchOpt = resolutions.find(resolution => resolution.maxDim == resPower)
+    if (snapToClosest) matchOpt.getOrElse(resolutions.minBy(resolution => math.abs(resolutionExponent - (math.log(resolution.maxDim)/math.log(2)))))
```

Comment:
Using the scale of the exponent to find the minimum seems absolutely correct to me. Yet this line is not clear at first sight. Some comments or refactoring would be helpful.

Reply:
To be honest, I’m not entirely sure why it is right to use the exponent rather than the actual value. I did so because it is used in selecting the right zoom step in the first place (in ImageThumbnail.scala ). Not sure how to refactor here, any suggestions?

Reply:
Let's have a look together tomorrow.